### ISSUE TITLE
Variable font version could not display some symbols

### ISSUE THREAD

**Issue by WordlessEcho** [adobe-fonts/source-han-sans#296]
### Prerequisites

* [x] If you are reporting an issue that affects glyphs for characters for a particular region or regions, did you verify that the characters are within the supported scope of the region or regions? This generally means GB 18030 or Tōngyòng Guīfàn Hànzìbiǎo (通用规范汉字表) for China, Big Five or CNS 11643 Planes 1 & 2 for Taiwan, HKSCS-2016 for Hong Kong, JIS X 0208, JIS X 0212, and JIS X 0213 for Japan, and KS X 1001 and KS X 1002 for Korea.
* [x] Did you thoroughly search the [open](https://github.com/adobe-fonts/source-han-sans/issues) and [closed](https://github.com/adobe-fonts/source-han-sans/issues?q=is%3Aissue+is%3Aclosed) issues to avoid reporting a duplicate issue?
* [x] Did you go through the [official font readme file](https://github.com/adobe-fonts/source-han-sans/raw/release/SourceHanSansReadMe.pdf) to better understand the scope of the project, to include the *Known Issues* section at the very end?

### Description

[Please provide here a thorough description of the issue, being sure to specify at least a Unicode code point if it affects a character or a particular glyph for a character. Including the character itself after the Unicode code point also helps, for visual reasons. Images with annotations that pinpoint the issue also help. If you are referencing standards, please keep in mind that they may have known or unknown errors. It is not necessary to provide CIDs.]

**Thanks to** @AokiFuru simonsmh/notocjk#25.

⚽ (U+26BD), ● (U+25CF), ■ (U+25A0), ◆ (U+25C6), ▲ (U+25B2), ▶ (U+25B6), ▼ (U+25BC), ◀ (U+25C0), ♠ (U+2660), ♥ (U+2665), ♣ (U+2663), ♦(U+2666), ❖ (U+2756), ✚ (U+271A), ✿ (U+273F), ◢ (U+25E2), ◣ (U+25E3), ◤ (U+25E4), ◥ (U+25E5)

Those were displayed correctly in super OTC version.

**Comment by RuixiZhang42** [comment]
@WordlessEcho 
> Those were displayed correctly in super OTC version.

English: No. The *Source Han* family doesn't cover these mentioned code points. Every font instance (including the super OTC) already reaches maximum glyph counts. The reason why these code points don't show up is because of other issues (maybe font fallback issue).

中文：思源家族（特指中日韩版本）本身就不带这些提到的字符，因为每个字体的字形量已经到达上限。之前能够显示，或许是因为字体回退机制可以正常运行（意思是这些字符是通过别的字体显示的，而不是思源）。现在，思源换作可变字体，可能暴露了字体回退机制的bug，而不是思源可变字体的bug。

**Comment by WordlessEcho** [comment]
I use Google's font tool to dump its cmap. ([googlefonts/fonttools: A library to manipulate font files from Python.](https://github.com/googlefonts/fonttools))

```xml
<cmap>
    <tableVersion version="0"/>
    <cmap_format_4 platformID="0" platEncID="3" language="0">
    <!-- some maps -->
    <map code="0x25c6" name="uni25C6"/><!-- BLACK DIAMOND -->
```

Now I am confused.

**Comment by punchcutter** [comment]
Those characters are in the font and look fine in some apps, but I see them disappear in other apps. What system and software are you using where they don't show up?

**Comment by punchcutter** [comment]
Actually, something very weird happened and a bunch of those glyph outlines disappeared. My local build is totally fine so we will be doing a new release soon that will fix this.

**Comment by WordlessEcho** [comment]
> Those characters are in the font and look fine in some apps, but I see them disappear in other apps. What system and software are you using where they don't show up?

**Windows 10 version 19042.867**
In System settings -> personalization -> fonts -> Source Han Sans HC VF -> input a ⚽ (U+26BD) for example
![image](https://user-images.githubusercontent.com/12007025/114531376-6ccb3080-9c7e-11eb-8b49-34d5e97f0b2d.png)

In Source Han Sans HC:
![image](https://user-images.githubusercontent.com/12007025/114531518-8d938600-9c7e-11eb-91c9-746f0745b5bd.png)

In Microsoft Edge (Chromium version), change font to Noto Sans CJK SC VF in settings -> appearence -> custom fonts
Some symbols fallback to use Emoji font correctly but some not.
![image](https://user-images.githubusercontent.com/12007025/114530251-596b9580-9c7d-11eb-9468-3be5b2fa27ed.png)

Change font-family by developer tools to Source Han Sans SC VF:
![image](https://user-images.githubusercontent.com/12007025/114530977-0e05b700-9c7e-11eb-923b-783bf6b24169.png)

Change font-family by developer tools to Source Han Sans SC:
![image](https://user-images.githubusercontent.com/12007025/114531159-38f00b00-9c7e-11eb-8001-3edcee01b484.png)

**Andorid 11**
I change system font config ([fonts.xml](https://android.googlesource.com/platform/frameworks/base/+/refs/heads/android11-release/data/fonts/fonts.xml)) to use variable font.
```xml
<!-- some family -->
<family lang="zh-Hans">
        <font weight="100" style="normal" index="3">NotoSansCJK-VF.ttc
            <axis tag="wght" stylevalue="100.0" />
        </font>
        <font weight="300" style="normal" index="3">NotoSansCJK-VF.ttc
            <axis tag="wght" stylevalue="300.0" />
        </font>
        <font weight="350" style="normal" index="3">NotoSansCJK-VF.ttc
            <axis tag="wght" stylevalue="350.0" />
        </font>
        <font weight="400" style="normal" index="3">NotoSansCJK-VF.ttc
            <axis tag="wght" stylevalue="400.0" />
        </font>
        <font weight="500" style="normal" index="3">NotoSansCJK-VF.ttc
            <axis tag="wght" stylevalue="500.0" />
        </font>
        <font weight="700" style="normal" index="3">NotoSansCJK-VF.ttc
            <axis tag="wght" stylevalue="700.0" />
        </font>
        <font weight="900" style="normal" index="3">NotoSansCJK-VF.ttc
            <axis tag="wght" stylevalue="900.0" />
        </font>

        <!-- serif font -->
    </family>
    <family lang="zh-Hant,zh-Bopo">
        <font weight="100" style="normal" index="4">NotoSansCJK-VF.ttc
            <axis tag="wght" stylevalue="100.0" />
        </font>
        <font weight="300" style="normal" index="4">NotoSansCJK-VF.ttc
            <axis tag="wght" stylevalue="300.0" />
        </font>
        <font weight="350" style="normal" index="4">NotoSansCJK-VF.ttc
            <axis tag="wght" stylevalue="350.0" />
        </font>
        <font weight="400" style="normal" index="4">NotoSansCJK-VF.ttc
            <axis tag="wght" stylevalue="400.0" />
        </font>
        <font weight="500" style="normal" index="4">NotoSansCJK-VF.ttc
            <axis tag="wght" stylevalue="500.0" />
        </font>
        <font weight="700" style="normal" index="4">NotoSansCJK-VF.ttc
            <axis tag="wght" stylevalue="700.0" />
        </font>
        <font weight="900" style="normal" index="4">NotoSansCJK-VF.ttc
            <axis tag="wght" stylevalue="900.0" />
        </font>

        <!-- serif font -->
    </family>
    <family lang="ja">
        <font weight="100" style="normal" index="1">NotoSansCJK-VF.ttc
            <axis tag="wght" stylevalue="100.0" />
        </font>
        <font weight="300" style="normal" index="1">NotoSansCJK-VF.ttc
            <axis tag="wght" stylevalue="300.0" />
        </font>
        <font weight="350" style="normal" index="1">NotoSansCJK-VF.ttc
            <axis tag="wght" stylevalue="350.0" />
        </font>
        <font weight="400" style="normal" index="1">NotoSansCJK-VF.ttc
            <axis tag="wght" stylevalue="400.0" />
        </font>
        <font weight="500" style="normal" index="1">NotoSansCJK-VF.ttc
            <axis tag="wght" stylevalue="500.0" />
        </font>
        <font weight="700" style="normal" index="1">NotoSansCJK-VF.ttc
            <axis tag="wght" stylevalue="700.0" />
        </font>
        <font weight="900" style="normal" index="1">NotoSansCJK-VF.ttc
            <axis tag="wght" stylevalue="900.0" />
        </font>

        <!-- serif font -->
    </family>
    <family lang="ko">
        <font weight="100" style="normal" index="2">NotoSansCJK-VF.ttc
            <axis tag="wght" stylevalue="100.0" />
        </font>
        <font weight="300" style="normal" index="2">NotoSansCJK-VF.ttc
            <axis tag="wght" stylevalue="300.0" />
        </font>
        <font weight="350" style="normal" index="2">NotoSansCJK-VF.ttc
            <axis tag="wght" stylevalue="350.0" />
        </font>
        <font weight="400" style="normal" index="2">NotoSansCJK-VF.ttc
            <axis tag="wght" stylevalue="400.0" />
        </font>
        <font weight="500" style="normal" index="2">NotoSansCJK-VF.ttc
            <axis tag="wght" stylevalue="500.0" />
        </font>
        <font weight="700" style="normal" index="2">NotoSansCJK-VF.ttc
            <axis tag="wght" stylevalue="700.0" />
        </font>
        <font weight="900" style="normal" index="2">NotoSansCJK-VF.ttc
            <axis tag="wght" stylevalue="900.0" />
        </font>

        <!-- serif font -->
    </family>
<!-- some family -->
```

In Google Pinyin Method
Before modified (Android 11 use NotoSansCJK-Regular.ttc version 2.001 in default):
![original](https://user-images.githubusercontent.com/12007025/114529829-eeba5a00-9c7c-11eb-935d-6e768d5ec482.png)

After modified:
![modified](https://user-images.githubusercontent.com/12007025/114529929-0bef2880-9c7d-11eb-9368-1920bf4ec185.png)

**Comment by tamcy** [comment]
@punchcutter In my case it is the autohinter that caused the disappearance of the glyphs. From my observation, the glyph would disappear when the outlines of the two masters are the same (but I do see a few exceptions).

**Comment by punchcutter** [comment]
@tamcy Thanks for confirming that. I recently found the bug in `tx` when used to instantiate a CFF2 variable font, but we aren't using that here. I just confirmed I see the same bug with `psautohint` since it also uses `tx` so looks like a bug in CFF2 handling.

**Comment by punchcutter** [comment]
Fixed in 2.004
